### ISSUE TITLE
Kirby Squeak Squad crashes when entering loading zone

### ISSUE THREAD

**Issue by SuperSonic472** [rafaelvcaetano/melonDS-android#1200]
* **Device model:** Moto G Stylus 5G
* **Android version:** Android 12 Build S2RES32.29-16-1-23

My phone does run all games at full FPS.

**Problem:**  
Pretty self explanatory, the game crashes when entering a loading zone like a door or warp star.

**Steps to reproduce (I recommend using the decrypted version from r/roms):**
1. Open a new save.
2. Enter any stage.
3. Enter a door.

**Video:**

https://github.com/rafaelvcaetano/melonDS-android/assets/64370131/35457a8d-6e73-4333-bffa-a27b6c00c187

**Comment by SuperSonic472** [comment]
Note: You might need to finish a few stages before the glitch happens.

**Comment by rafaelvcaetano** [comment]
Could you upload a save file that allows this issue to be easily reproduced?

**Comment by SuperSonic472** [comment]
I don't think it needs a save but here
[Download](https://drive.google.com/file/d/1_RLwlF8BgRX9nuaduRvRaiErDGKZCx8h/view?usp=sharing)

**Comment by rafaelvcaetano** [comment]
Just attaching some error logs related to this.

```
2023-08-27 18:17:55.713 14308-14308 DEBUG                   pid-14308                            A  pid: 14843, tid: 12793, name: Thread-24  >>> me.magnum.melonds.dev <<<
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #02 pc 000000000042dc50  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (ARMJIT::Compiler::Comp_MemAccess(int, int, ARMJIT::Op2, int, int)+2412) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #03 pc 000000000042dffc  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (ARMJIT::Compiler::T_Comp_LoadPCRel()+116) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #04 pc 0000000000429360  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (ARMJIT::Compiler::CompileBlock(ARM*, bool, ARMJIT::FetchedInstr*, int, bool)+952) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #05 pc 000000000040edf8  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (ARMJIT::CompileBlock(ARM*)+4376) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #06 pc 000000000039c328  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (ARMv5::ExecuteJIT()+480) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #07 pc 00000000003ea12c  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (unsigned int NDS::RunFrame<true, 0>()+532) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #08 pc 00000000003e9b0c  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (NDS::RunFrame()+72) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #09 pc 0000000000382470  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-lib.so (MelonDSAndroid::loop()+32) (BuildId: a48c252da952e7f639cd575fb51cc3612e4747fa)
2023-08-27 18:17:55.714 14308-14308 DEBUG                   pid-14308                            A        #10 pc 0000000000012048  /data/app/~~lmaPHrobBbuqNOF4I2JcoA==/me.magnum.melonds.dev-_szqfBlC0qDNfLOr2nQbsw==/lib/arm64/libmelonDS-android-frontend.so (emulate(void*)+284) (BuildId: f3c2742dada2dc26e0c9e394bbc276ce6b8f0fc3)
```

**Comment by rafaelvcaetano** [comment]
This should be fixed by commit [395be1d](https://github.com/rafaelvcaetano/melonDS-android/commit/395be1de8b0208108088ffc780220184c6c668ac). Try the latest nightly release once it's ready and let me know if it's fixed 🙂
